libs/src/services/ethContracts/index.js

```diff
@@ -185,41 +185,51 @@ class EthContracts {
-        }
+        try {
+          const {
+            data: { service: serviceName, version: serviceVersion }
+          } = await axios(
+            { url: urlJoin(sp.endpoint, 'version'), method: 'get' }
+          )
+          if (serviceName !== spType) {
+            throw new Error(`Invalid service type: ${serviceName}. Expected ${spType}`)
+          }
 
-        if (serviceName !== spType) {
-          throw new Error(`Invalid service type: ${serviceName}. Expected ${spType}`)
-        }
+          if (!semver.valid(serviceVersion)) {
+            throw new Error(`Invalid semver version found - ${serviceVersion}`)
+          }
 
-        if (!semver.valid(serviceVersion)) {
-          throw new Error(`Invalid semver version found - ${serviceVersion}`)
-        }
+          if (expectedVersion !== serviceVersion) {
+            // Confirm this returned value is valid, ignoring patch version in semantic versioning
+            let validSPVersion = this.isValidSPVersion(expectedVersion, serviceVersion)
+            if (!validSPVersion) {
+              throw new Error(`Invalid latest service version: ${serviceName}. Expected ${expectedVersion}, found ${serviceVersion}`)
+            }
+          }
 
-        if (expectedVersion !== serviceVersion) {
-          // Confirm this returned value is valid, ignoring patch version in semantic versioning
-          let validSPVersion = this.isValidSPVersion(expectedVersion, serviceVersion)
-          if (!validSPVersion) {
-            throw new Error(`Invalid latest service version: ${serviceName}. Expected ${expectedVersion}, found ${serviceVersion}`)
+          // Discovery provider specific validation
+          if (spType === 'discovery-provider') {
+            await this.validateDiscoveryProviderHealth(sp.endpoint)
           }
-        }
 
-        foundVersions.add(serviceVersion)
-        // Update mapping of version <-> [endpoint], creating array if needed
-        if (!spVersionToEndpoint.hasOwnProperty(serviceVersion)) {
-          spVersionToEndpoint[serviceVersion] = [sp.endpoint]
-        } else {
-          spVersionToEndpoint[serviceVersion].push(sp.endpoint)
+          foundVersions.add(serviceVersion)
+          // Update mapping of version <-> [endpoint], creating array if needed
+          if (!spVersionToEndpoint.hasOwnProperty(serviceVersion)) {
+            spVersionToEndpoint[serviceVersion] = [sp.endpoint]
+          } else {
+            spVersionToEndpoint[serviceVersion].push(sp.endpoint)
+          }
+        } catch (e) {
+          // Swallow errors for a single sp endpoint to ensure others can proceed
+          console.error(`Failed to retrieve information for ${sp}`)
         }
       }))
       let foundVersionsList = Array.from(foundVersions)
+      if (foundVersionsList.length === 0) {
+        // Short-circuit processing if no valid endpoints are found
+        throw new Error(`No valid endpoints found for ${spType}`)
```

Comment:
@hareeshnagaraj i don't think throwing an error is what we want here. if this returns null doesn't it try the previous version?

Reply:
that is what we want - an error exits and sends us to selecting the previous version, since no valid versions were found in this set